feat(core): add conflict handling

tlvince opened the pull request:
Recursively declare the remote doc the winner if it's newer.

How to test this:

1. Make sure the app is not running
2. In CouchDB, set the OPV packed quantity of a daily delivery doc to 999 and
   set the modified date to a future date (e.g. an hour ahead)
3. Save the doc
4. Kill CouchDB
5. Load the app
6. Set the packed quantity of OPV to 1
7. Save the doc
8. Close the app
9. Start up CouchDB
10. Start up the app

After sync, OPV packed quantity should be 999.

Closes #255.

jofomah commented:
- This currently checks against all doc types. Should it?
  Yes, but for some doc_types, remote doc should always win, e.g product type and other documents that does not get modified on the tablet.

Its currently silent. Should we inform the user?
 Will be nice if we pop up an alert telling the user that conflict have been resolved, will help in case of troubleshooting, i think so

The conflicting doc(s) will be marked for deletion. Should they?
Currently on the app, it runs compaction so in most cases, only the latest version or copy exists


About remote document winning if it is newer, what if a document was modified like in the scenario you gave but after a delivery have been made, IMO i think it is of no use updating the local copy anymore since the delivery have been made already.

tlvince commented:
> This currently checks against all doc types. Should it?

I've added whitelisting; remote will win unless the doc type is in the whitelist.

> Its currently silent. Should we inform the user?

Added toasts.

> The conflicting doc(s) will be marked for deletion. Should they?

They will be marked for deletion on the server.

> About remote document winning if it is newer […]

Agreed with that scenario. Will add it.

tlvince commented:
@jofomah I think this is ready for testing.
